### ISSUE TITLE
several `Status` in OPs are not checked

### ISSUE THREAD

**Issue by njzjz** [deepmodeling/deepmd-kit#2513]
### Summary

The compilation warnings are raised.

They have the risk of segment fault due to not being checked.

### DeePMD-kit Version

devel

### TensorFlow Version

2.12.0

### Details

```
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc: In instantiation of 'int _norm_copy_coord_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, FPTYPE*&, int*&, int*&, int&, int&, const FPTYPE*, const FPTYPE*, const int*, const int&, const int&, const float&) [with FPTYPE = double]':
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1644:41:   required from 'void _prepare_coord_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, const FPTYPE**, FPTYPE*&, const int**, int*&, int*&, deepmd::InputNlist&, int*&, int*&, int**&, int*&, int*&, int&, int&, int&, int&, const FPTYPE*, const int*, int, const int&, const int&, const float&, const int&, const int&) [with FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1182:41:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::_Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1005:59:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1003:8:   required from here
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1481:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1481 |   context->allocate_temp(DataTypeToEnum<FPTYPE>::value, FPTYPE_shape,
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1499:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1499 |   context->allocate_temp(DataTypeToEnum<FPTYPE>::value, double_shape,
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1523:5: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1523 |     context->allocate_temp(DataTypeToEnum<FPTYPE>::value, cpy_shape,
      |     ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc: In instantiation of 'int _build_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, int*&, int*&, int**&, int*&, int&, int&, const FPTYPE*, const int&, const int&, const int&, const float&) [with FPTYPE = double]':
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1654:25:   required from 'void _prepare_coord_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, const FPTYPE**, FPTYPE*&, const int**, int*&, int*&, deepmd::InputNlist&, int*&, int*&, int**&, int*&, int*&, int&, int&, int&, int&, const FPTYPE*, const int*, int, const int&, const int&, const float&, const int&, const int&) [with FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1182:41:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::_Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1005:59:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = double]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1003:8:   required from here
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1563:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1563 |   context->allocate_temp(DT_INT32, nlist_shape, tensor_list);
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1574:5: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1574 |     context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
      |     ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc: In instantiation of 'int _norm_copy_coord_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, FPTYPE*&, int*&, int*&, int&, int&, const FPTYPE*, const FPTYPE*, const int*, const int&, const int&, const float&) [with FPTYPE = float]':
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1644:41:   required from 'void _prepare_coord_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, const FPTYPE**, FPTYPE*&, const int**, int*&, int*&, deepmd::InputNlist&, int*&, int*&, int**&, int*&, int*&, int&, int&, int&, int&, const FPTYPE*, const int*, int, const int&, const int&, const float&, const int&, const int&) [with FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1182:41:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::_Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1005:59:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1003:8:   required from here
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1481:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1481 |   context->allocate_temp(DataTypeToEnum<FPTYPE>::value, FPTYPE_shape,
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1499:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1499 |   context->allocate_temp(DataTypeToEnum<FPTYPE>::value, double_shape,
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1523:5: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1523 |     context->allocate_temp(DataTypeToEnum<FPTYPE>::value, cpy_shape,
      |     ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc: In instantiation of 'int _build_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, int*&, int*&, int**&, int*&, int&, int&, const FPTYPE*, const int&, const int&, const int&, const float&) [with FPTYPE = float]':
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1654:25:   required from 'void _prepare_coord_nlist_gpu(tensorflow::OpKernelContext*, tensorflow::Tensor*, const FPTYPE**, FPTYPE*&, const int**, int*&, int*&, deepmd::InputNlist&, int*&, int*&, int**&, int*&, int*&, int&, int&, int&, int&, const FPTYPE*, const int*, int, const int&, const int&, const float&, const int&, const int&) [with FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1182:41:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::_Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1005:59:   required from 'void ProdEnvMatAMixOp<Device, FPTYPE>::Compute(tensorflow::OpKernelContext*) [with Device = Eigen::GpuDevice; FPTYPE = float]'
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1003:8:   required from here
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1563:3: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1563 |   context->allocate_temp(DT_INT32, nlist_shape, tensor_list);
      |   ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
/root/deepmd-kit/source/op/prod_env_mat_multi_device.cc:1574:5: warning: ignoring returned value of type 'tsl::Status', declared with attribute nodiscard [-Wunused-result]
 1574 |     context->allocate_temp(DT_INT32, jlist_shape, tensor_list + 1);
      |     ^~~~~~~
In file included from /root/deepmd-kit/source/op/custom_op.h:8,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_kernel.h:1006:10: note: in call to 'tsl::Status tensorflow::OpKernelContext::allocate_temp(tensorflow::DataType, const tensorflow::TensorShape&, tensorflow::Tensor*)', declared here
 1006 |   Status allocate_temp(DataType type, const TensorShape& shape,
      |          ^~~~~~~~~~~~~
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/platform/status.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/lib/core/status.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_base.h:20,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/resource_handle.h:21,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:27,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:23,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from /root/deepmd-kit/source/op/custom_op.h:7,
                 from /root/deepmd-kit/source/build_c/op/CMakeFiles/deepmd_op.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/tsl/platform/status.h:63:7: note: 'tsl::Status' declared here
   63 | class Status {
      |       ^~~~~~
```
